codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -138,6 +139,21 @@ class ServerProtocolTestGenerator(
         }
     }
 
+    private fun List<TestCase>.fixBroken(): List<TestCase> = this.map { when (it) {
+        is TestCase.RequestTest -> {
+            val howToFixIt = BrokenRequestTests[Pair(codegenContext.serviceShape.id.toString(), it.testCase.id)]
+            if (howToFixIt == null) {
+                it
+            } else {
+                val fixed = howToFixIt(it.testCase)
+                TestCase.RequestTest(fixed, it.targetShape)
+            }
+        }
+        is TestCase.ResponseTest -> {
+            it
+        }
+    } }
```

Comment:
Probably a comment would help here. It is clear in the PR, but it will be come harder when it is just part of a big kotlin source file like this one.